homeassistant/components/daikin_madoka/climate.py

```diff
@@ -0,0 +1,363 @@
+    OperationModeEnum.FAN: HVAC_MODE_FAN_ONLY,
+    OperationModeEnum.DRY: HVAC_MODE_DRY,
+    OperationModeEnum.COOL: HVAC_MODE_COOL,
+    OperationModeEnum.HEAT: HVAC_MODE_HEAT,
+    OperationModeEnum.AUTO: HVAC_MODE_AUTO,
+}
+
+HA_FAN_MODE_TO_DAIKIN = {
+    FAN_LOW: FanSpeedEnum.LOW,
+    FAN_MEDIUM: FanSpeedEnum.MID,
+    FAN_HIGH: FanSpeedEnum.HIGH,
+    FAN_AUTO: FanSpeedEnum.AUTO,
+}
+
+DAIKIN_TO_HA_FAN_MODE = {
+    FanSpeedEnum.LOW: FAN_LOW,
+    FanSpeedEnum.MID: FAN_MEDIUM,
+    FanSpeedEnum.HIGH: FAN_HIGH,
+    FanSpeedEnum.AUTO: FAN_AUTO,
+}
+
+DAIKIN_TO_HA_CURRENT_HVAC_MODE = {
+    OperationModeEnum.FAN: CURRENT_HVAC_FAN,
+    OperationModeEnum.DRY: CURRENT_HVAC_DRY,
+    OperationModeEnum.COOL: CURRENT_HVAC_COOL,
+    OperationModeEnum.HEAT: CURRENT_HVAC_HEAT,
+}
+
+DATA = "data"
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Daikin climate based on config_entry."""
+
+    entities = []
+
+    for controller in hass.data[DOMAIN][entry.entry_id][CONTROLLERS].values():
+        try:
+            entity = DaikinMadokaClimate(controller)
+            entities.append(entity)
+            await entity.controller.update()
+        except ConnectionAbortedError:
+            pass
+        except ConnectionException:
+            pass
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class DaikinMadokaClimate(ClimateEntity):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, controller: Controller):
+        """Initialize the climate device."""
+        self.controller = controller
+        self.dev_info = None
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def available(self):
+        """Return the availability."""
+        return (
+            self.controller.connection.connection_status == ConnectionStatus.CONNECTED
+        )
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return (
+            self.controller.connection.name
+            if self.controller.connection.name is not None
+            else self.controller.connection.address
+        )
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self.controller.connection.address
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self.controller.temperatures.status is None:
+            return MIN_TEMP
+
+        return self.controller.temperatures.status.indoor
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+
+        if self.controller.set_point.status is None:
+            return MIN_TEMP
```

Comment:
Is it always set? Why set it to MIN_TEMP instead of None, when no target_temperature is set?

Reply:
Updated. Somehow it made it easier to test with the fixtures, but I replaced the values for json null and can handle empty state accordingly. Thanks!